library/general/src/lib/cfa/sysctl_config.rb

```diff
@@ -0,0 +1,211 @@
+
+module CFA
+  # CFA based API to adjust the sysctl tool configuration
+  #
+  # This class does not modify the running kernel configuration. It just writes
+  # the desired values into the configuration file ({PATH}).
+  # Despite the class Sysctl this class also takes care about entries in
+  #   /run/sysctl.d,
+  #   /etc/sysctl.d
+  #   /usr/local/lib/sysctl.d
+  #   /usr/lib/sysctl.d
+  #   /lib/sysctl.d
+  #   /etc/sysctl.conf
+  # and inform the user if his settings will be overruled by setting in
+  # other files.
+  #
+  # @example Enabling IPv4 forwarding
+  #   sysctl = SysctlConfig.new
+  #   sysctl.forward_ipv4 = true
+  #   sysctl.save
+  #
+  # Although in the previous example we enabled the IPv4 forwarding using by
+  # setting +forward_ipv4+ to true. However, under the hood, the kernel maps
+  # boolean values to "1" or "0". If you want to access to that raw value,
+  # you can prepend "raw_" to the method's name.
+  #
+  # @example Accessing the raw value of a setting
+  #   sysctl = SysctlConfig.new
+  #   sysctl.load
+  #   sysctl.raw_forward_ipv6 #=> "0"
+  #   sysctl.raw_forward_ipv6 = "1"
+  class SysctlConfig
+    include Yast::Logger
+    include Yast::I18n
+    extend Yast::I18n
+
+    textdomain "base"
+
+    PATHS = [
+      "/run/sysctl.d",
+      "/etc/sysctl.d",
+      "/usr/local/lib/sysctl.d",
+      "/usr/lib/sysctl.d",
+      "/lib/sysctl.d",
+      "/etc/sysctl.conf"
+    ].freeze
+    private_constant :PATHS
+
+    YAST_CONFIG_PATH = Sysctl::PATH
+    private_constant :YAST_CONFIG_PATH
+
+    class << self
+      def define_attr(attr)
+        define_method attr do
+          file = files.reverse.find do |f|
+            f.present?(attr)
+          end
+          return file.public_send(attr) if file
+
+          yast_config_file.public_send(attr)
+        end
+
+        define_method "#{attr}=" do |value|
+          yast_config_file.public_send("#{attr}=", value)
+        end
+      end
+    end
+
+    Sysctl.known_attributes.each { |a| define_attr(a) }
+
+    def load
+      files.each(&:load)
+    end
+
+    # Saving all sysctl settings
+    #
+    # @param check_conflicts [Boolean] checking if the settings are overruled
+    def save(check_conflicts: true)
+      if check_conflicts
+        conflict_files = conflict_files()
+        if !conflict_files.empty?
+          Yast::Report.Warning(format(_("The settings have been written to %{yast_file_name}.\n"\
+            "But they will be overruled be manual setting described in %{file_list}."), yast_file_name: YAST_CONFIG_PATH, file_list: conflict_files.join(", ")))
+        end
+      end
+      yast_config_file&.save
+    end
+
+    # Whether there is a conflict with given attributes
+    #
+    # @param only [Array<Symbol>] attributes to check
+    # @return [Array<String>] list of conflicting files
+    def conflict_files(only: [])
+      return [] if yast_config_file.empty?
+
+      conflicting_attrs = yast_config_file.present_attributes
```

Comment:
I would delegate to `CFA::Sysctl` the comparison between two files. Again, [CFA:LoginDefs](https://github.com/yast/yast-yast2/blob/master/library/general/src/lib/cfa/login_defs.rb#L127) does something similar.

Reply:
Nice, good idea.....